apis/python/src/tiledb/vector_search/object_readers/directory_reader.py

```diff
@@ -178,7 +409,18 @@ def read_objects(
                 try:
                     local_temp_path = ""
                     parsed_path = urlparse(path, allow_fragments=False)
-                    if parsed_path.scheme != "file":
+                    if parsed_path.scheme == "tiledb":
+                        key = parsed_path.path.lstrip("/")
+                        with tiledb.open(path, "r") as a:
+                            file_extension = a.meta["file_extension"].lstrip(".")
+                            file_size = a.meta["file_size"]
+                            local_temp_path = f"{temp_dir}/{key}.{file_extension}"
+                            os.makedirs(os.path.dirname(local_temp_path), exist_ok=True)
+                            data = a[:file_size]["contents"]
+                            with open(local_temp_path, "wb") as dst:
+                                dst.write(data)
+                        parsed_path = urlparse(local_temp_path, allow_fragments=False)
+                    if parsed_path.scheme == "s3":
```

Comment:
Can we do the same with s3 to avoid copying? Copying anything should be avoided and removed if possible.

Reply:
See comment above